plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -440,6 +440,31 @@ func ClusterRoles() []rbacv1.ClusterRole {
 				rbacv1helpers.NewRule(Read...).Groups(legacyGroup).Resources("persistentvolumeclaims", "persistentvolumes").RuleOrDie(),
 			},
 		},
+		{
+			// a role for the Cluster Autoscaler
+			ObjectMeta: metav1.ObjectMeta{Name: "system:cluster-autoscaler"},
```

Comment:
where does the cluster autoscaler component live? this seems like it would be better defined in a manifest defined in https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler

Reply:
At least in GCE & GKE the CA lives on master node and is started from the manifest file that resides in kubernetes/kubernetes repository and is part of this PR - please have a look at [cluster-autoscaler.manifest](https://github.com/kubernetes/kubernetes/pull/64503/files#diff-362fc8c6835ec66fba8c84b85e29e6f0).

Reply:
since the component is deployment-specific, the permissions for the autoscaler should also be part of the deployment. for an example of what this could look like for GCE/GKE, see the permissions GCE/GKE grants to kubelets:

https://github.com/kubernetes/kubernetes/blob/bb869d83890edf35c4200cf50651e68b8c9cba49/cluster/gce/gci/configure-helper.sh#L2246-L2248

https://github.com/kubernetes/kubernetes/blob/bb869d83890edf35c4200cf50651e68b8c9cba49/cluster/addons/rbac/kubelet-cert-rotation/kubelet-certificate-management.yaml#L1-L62